ocaml_emulator/riscv_ocaml_sim.ml

```diff
@@ -109,13 +116,13 @@ let check_elf () =
 let write_bytes fl bytes =
   let i = ref 0 in
   while !i < Bytes.length bytes do
-    let s = Printf.sprintf "%02x%02x%02x%02x\n"
-	(int_of_char (Bytes.get bytes (!i+3)))
-	(int_of_char (Bytes.get bytes (!i+2)))
-	(int_of_char (Bytes.get bytes (!i+1)))
-	(int_of_char (Bytes.get bytes (!i+0))) in
-    output_string fl s;
-    i := !i + 4
+    for j = 0 to !signature_granularity_ref - 1 do
+      let s = Printf.sprintf "%02x"
+          (int_of_char (Bytes.get bytes (!i + (!signature_granularity_ref - 1 - j)))) in
```

Comment:
You could use `1 to !signature_granularity_ref` for the loop and drop the - 1 here, or just `for j = !signature_granularity_ref - 1 downto 0` like the C version and not have to do the inversion at all